internal/app/machined/pkg/runtime/v1alpha1/platform/metal/internal/url/url.go

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package url handles expansion of the download URL for the config.
+package url
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/gen/slices"
+)
+
+// Populate populates the config download URL with values replacing variables.
+func Populate(ctx context.Context, downloadURL string, st state.State) (string, error) {
+	return PopulateVariables(ctx, downloadURL, st, AllVariables())
```

Comment:
same function signatures? and I don;t see `PopulateVariables` being used as is :thinking: ?

Reply:
I kind of did more generic version which accepts custom variable list...